accessibility/README.md

```diff
@@ -0,0 +1,490 @@
+* [Ensure correct usage of `aria-labelledby` and `aria-label`](#ensure-correct-usage-of-aria-labelledby-and-aria-label)
+
+### Ensure that `input` elements have corresponding `label` elements
+
+#### Don't
+
+```html
+<input
+  id="newsletter-email"
+  name="email" 
+  type="email"
+  placeholder="Enter your email"
+>
+```
+
+#### Do
+
+```html
+<label for="newsletter-email">
+  Enter your email
+</label>
+
+<input
+  id="newsletter-email"
+  name="email" 
+  type="email"
+  placeholder="Enter your email"
+>
+```
+
+* Ensuring that you have that an `input` has a corresponding `label` element helps to describe the purpose and can provide instructions for the input for users
+* A `label` should be used even when the input has a `placeholder` attribute.
+  * Assistive technology, such as screen-readers do not treat placeholder text as labels, so a `placeholder` should be used alongside a label, not instead of a label
+* In some cases, a site's design may show inputs without visible labels. Sometimes this may be because other elements on the page give the input some context - for example a search text input placed directly next to a "Search" button.
+  * If a input does not have a visible label, a `<label>` element should still be added for the benefit of screen readers
+  * A label can be hidden visually by adding the `visually-hidden` class to the `label`
+  ```html
+  <!-- Label will be available to assistive technology, but visually hidden -->
+  <label class="visually-hidden" for="search">
+    Search the site
+  </label>
+
+  <input id="search" type="text" name="search">
+  ```
+
+### Ensure interactive elements have accessible text
+
+#### Don't
+
+```vue
+<a class="site-logo" href="/">
+  <brand-logo />
+</a>
+
+<button
+  type="button"
+  @click="openMenuDrawer"
+>
+  <icon-menu />
+</button>
+
+<a href="/account">
+  <icon-account />
+</a>
+
+<a href="/sale">
+  <!-- Image containing embedded text about a 10% off sale -->
+  <img src="10-off-promo-image.jpg">
+</a>
+
+
+```
+
+#### Do
+
+```vue
+<a class="site-logo" href="/">
+  <brand-logo />
+  
+  <span class="visually-hidden">
+    My Shop Name
+  </span>
+</a>
+
+<button
+  type="button"
+  @click="openMenuDrawer"
+>
+  <icon-menu />
+  
+  <span class="visually-hidden">
+    Toggle Menu
+  </span>
+</button>
+
+<a href="/account">
+  <icon-account />
+  
+  <span class="visually-hidden">
+    Account
+  </span>
+</a>
+
+<a href="/sale">
+  <!-- Image containing embedded text about a 10% off sale -->
+  <img src="10-off-promo-image.jpg" alt="Get 10% off">
+</a>
+```
+
+* When using `<button>` or `<a>` elements containing SVG elements, ensure that accessible text is also included
+  * If the text should not be visible, the `visually-hidden` class can be used
+* If a `<button>` or `<a>` contains a `<img>` element - ensure that there is either appropriate `alt` text on the image, or add a supporting `visually-hidden` span with appropriate text
```

Comment:
Would it be possible to add a note that if `visually-hidden` text is used, the `alt` should be empty? Otherwise screenreaders will read out possibly duplicate or irrelevant content

Reply:
Good point - I've added a line about this